### DISCUSSION TITLE
Allow assignment of items for tracking loans / checkout etc

### DISCUSSION THREAD

**Discussion by IamSAB** [sysadminsmedia/homebox#713] [Ideas]
### What is the problem you are trying to solve with this feature?

Our association runs a materials warehouse. People can borrow items or we pick out items for a camp or event.
I must know what items quantities are borrowed or picked out.

### What is the solution you are proposing?

We can use existing sytem like:
- PersonA borrows two ropes from LocationA to go climb -> entry: quantity=2, item=rope, location=Address of PersonA, label=PersonA
- PersonB picks out two tents from LocationB for a summer camp -> entry: quantity=2, item=tent, location=Adress of Camp, label=PersonB
- PersonC reserves five tents from LocationC for a summer camp -> entry: quantity=5, item=tent, location=LocationC, label=PersonC

But would require to add
- a m2m with quantity and label pivot between items and location.
- better bulk editing of items (for  location and labels)

### What alternatives have you considered?

Add a new table which lists borrowed or checked out items, a new feature. Would be more versatile.

_However, I would anyway prefer to be able to store same items at different locations. That's also depends on the granularity of the "location". If a location is not only a room, but a specific box on a shelf in a room than its quite limiting to be not able to assign items to various locations._

### Contributions

- [x] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [ ] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [x] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by katosdev** [comment]
REF: https://discord.com/channels/1254144480391598163/1292711847895629895

**Comment by jgillman** [comment]
I have a similar but possibly more "home-related" use case: Sometimes I lend my stuff out to friends and family and I would like to keep track of who I've lend it to and when.

I'm also open to feedback if there's an existing way to do this? Maybe with labels? Or people-as-locations?

Let me know if I should break this out to a separate issue/enhancement.

**Comment by LukasPavlas** [comment]
if it was possible to add a user manually and not allow him to log in. Then assign it to the asset as a label

**Comment by megavolts** [comment]
This features would be awesome, and really improve homebox. I've been looking at other inventory alternative that will allow to checkout item for a loan, but either they lack the simplicity of homebox, or are design for specific purpose (e.g. snipe-it).

**Comment by tankerkiller125** [comment]
Please note we're converting this issue into a discussion as per our new procedures/policies. You can read more about that in our announcement #712

**Comment by simonwiles** [comment]
This is the main feature I would want HomeBox for.  I think it could fit elegantly with HomeBox's feature set and existing use-cases and significantly increase HomeBox's range of applicability without substantially increasing complexity or requiring additional architecture.

**Comment by sharmay** [comment]
Just like the "sold", if it can be a drop down ex Loaned, Borrowed, Donated would be a very good addition. To track the item status.

**Comment by ofreiheit** [comment]
I would be really interested in such a feature. Ideally, it would have a history so I can see who I have lent it to over time.

**Comment by jtatknox** [comment]
I think this would be  a great feature, too. Simplest implementation might just be check out mode that can be toggled to check in mode. I am imagining a simple use case of moving my stuff in boxes and scanning as they are loaded on and off a moving truck.

**Comment by Saturo78** [comment]
I use locations to represent people. When an item is lent out, I simply change its location to the person who has it. That clearly answers the most important question: who currently has the item. Locations are already a core concept in HomeBox, searchable, filterable, and visible everywhere, so this feels natural and requires no extra logic.

For the lending state itself, I use tags like “borrowed” or “on loan”. This makes the status immediately visible and allows easy filtering of all lent-out items. Changing the status is quick and works well with bulk edits.

**Reply by ofreiheit** [reply]
It's an option, but it quickly becomes confusing as soon as you have a large group of people to whom you are lending equipment.
Also, there is no history, so it's difficult to track if any accessories get lost from one person to another.

If I could vote, I would like to suggest adding an option for checking out or borrowing equipment.
